BookShelfBusinessLogic/LibraryService.cs

```diff
@@ -0,0 +1,379 @@
+            _db.SaveChanges();
+
+            return true;
+        }
+
+        /// <summary>
+        /// Tries to add author to the specified book.
+        /// </summary>
+        /// <param name="bookId"> Id of book to be updated </param>
+        /// <param name="authorId"> Id of author to be added </param>
+        /// <returns> True if updated, false if such author already added to this book or such author or book does not exists </returns>
+        public bool IsAuthorToBookAdded(int bookId, int authorId)
+        {
+            BookAuthor bookAuthor = new BookAuthor { BookRefId = bookId, AuthorRefId = authorId };
+            BookAuthor similarEntry = _db.BookAuthor.FirstOrDefault(ba => ba.BookRefId == bookId && ba.AuthorRefId == authorId);
+            if (similarEntry != null)
+            {
+                return false;
+            }
+
+            _db.BookAuthor.Add(bookAuthor);
+            _db.SaveChanges();
+
+            return true;
+        }
+
+        /// <summary>
+        /// Tries to remove genre from the specified book.
+        /// </summary>
+        /// <param name="bookId"> Id of book to be updated </param>
+        /// <param name="genreId"> Id of genre to be removed </param>
+        /// <returns> True if updated, false if such genre has not been added to this book previously </returns>
+        public bool IsGenreFromBookRemoved(int bookId, int genreId)
+        {
+            BookGenre bookGenreToRemove = _db.BookGenre.FirstOrDefault(bg => bg.BookRefId == bookId && bg.GenreRefId == genreId);
+
+            if (bookGenreToRemove == null)
+            {
+                return false;
+            }
+
+            _db.BookGenre.Remove(bookGenreToRemove);
+            _db.SaveChanges();
+
+            return true;
+        }
+
+
+        /// <summary>
+        /// Tries to remove author from the specified book.
+        /// </summary>
+        /// <param name="bookId"> Id of book to be updated </param>
+        /// <param name="authorId"> Id of author to be removed </param>
+        /// <returns> True if updated, false if such author has not been added to this book previously </returns>
+        public bool IsAuthorFromBookRemoved(int bookId, int authorId)
+        {
+            BookAuthor bookAuthorToRemove = _db.BookAuthor.FirstOrDefault(ba => ba.BookRefId == bookId && ba.AuthorRefId == authorId);
+
+            if (bookAuthorToRemove == null)
+            {
+                return false;
+            }
+
+            _db.BookAuthor.Remove(bookAuthorToRemove);
+            _db.SaveChanges();
+
+            return true;
+        }
+
+        /// <summary>
+        /// Selects all Books by specified author
+        /// </summary>
+        /// <param name="authorId">Id of author</param>
+        /// <returns>The collection of Books </returns>
+        public IEnumerable<Book> GetBooksByAuthor(int authorId)
+        {
+            IEnumerable<Book> booksByAuthor = from book in _db.Books
```

Comment:
I don't know you used sql syntax because it is easier or it was a plan, any way FYI
https://stackoverflow.com/questions/2767709/join-where-with-linq-and-lambda